File: system-test/java/src/main/java/com/optivem/shop/testkit/port/given/steps/GivenProduct.java
package com.optivem.shop.testkit.port.given.steps;

import com.optivem.shop.testkit.port.given.steps.base.GivenStep;

public interface GivenProduct extends GivenStep {
    GivenProduct withSku(String sku);

    GivenProduct withUnitPrice(String unitPrice);

    GivenProduct withUnitPrice(double unitPrice);
}

File: system-test/typescript/src/testkit/dsl/port/given/steps/given-product.ts
import type { GivenStage } from '../given-stage.js';
import type { WhenStage } from '../../when/when-stage.js';
import type { ThenStage } from '../../then/then-stage.js';

export interface GivenProduct {
  withSku(sku: string): this;
  withUnitPrice(unitPrice: string | number): this;
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}
